tests/integration/test_rex.py

```diff
@@ -1121,3 +1107,134 @@ def test_extract_date(c, gpu):
         {"i": [datetime(2021, 1, 3), datetime(2022, 2, 4), datetime(2023, 3, 5)]}
     )
     assert_eq(result, expected_df)
+
+
+@pytest.mark.parametrize(
+    "gpu",
+    [
+        False,
+        pytest.param(
+            True,
+            marks=(
+                pytest.mark.gpu,
+                pytest.mark.xfail(
+                    not DASK_CUDF_TODATETIME_SUPPORT,
+                    reason="Requires https://github.com/dask/dask/pull/9881",
+                    raises=RuntimeError,
+                ),
+            ),
+        ),
+    ],
+)
+def test_scalar_timestamps(c, gpu):
+    df = pd.DataFrame({"d": [1203073300, 1503073700]})
+    c.create_table("df", df, gpu=gpu)
+
+    expected_df = pd.DataFrame(
+        {
+            "dt": [datetime(2008, 2, 20, 11, 1, 40), datetime(2017, 8, 23, 16, 28, 20)],
+        }
+    )
+
+    df1 = c.sql("SELECT to_timestamp(d) + INTERVAL '5 days' AS dt FROM df")
+    assert_eq(df1, expected_df)
+    df2 = c.sql("SELECT CAST(d AS TIMESTAMP) + INTERVAL '5 days' AS dt FROM df")
+    assert_eq(df2, expected_df)
+
+    df1 = c.sql("SELECT TIMESTAMPADD(DAY, 5, to_timestamp(d)) AS dt FROM df")
+    assert_eq(df1, expected_df)
+    df2 = c.sql("SELECT TIMESTAMPADD(DAY, 5, d) AS dt FROM df")
+    assert_eq(df2, expected_df)
+    df3 = c.sql("SELECT TIMESTAMPADD(DAY, 5, CAST(d AS TIMESTAMP)) AS dt FROM df")
+    assert_eq(df3, expected_df)
+
+    expected_df = pd.DataFrame({"day": [15, 18]})
+    df1 = c.sql("SELECT EXTRACT(DAY FROM to_timestamp(d)) AS day FROM df")
+    assert_eq(df1, expected_df, check_dtype=False)
+    df2 = c.sql("SELECT EXTRACT(DAY FROM CAST(d AS TIMESTAMP)) AS day FROM df")
+    assert_eq(df2, expected_df, check_dtype=False)
+
+    expected_df = pd.DataFrame(
+        {
+            "ceil_to_day": [datetime(2008, 2, 16), datetime(2017, 8, 19)],
+        }
+    )
+    df1 = c.sql("SELECT CEIL(to_timestamp(d) TO DAY) AS ceil_to_day FROM df")
+    assert_eq(df1, expected_df)
+    df2 = c.sql("SELECT CEIL(CAST(d AS TIMESTAMP) TO DAY) AS ceil_to_day FROM df")
+    assert_eq(df2, expected_df)
+
+    expected_df = pd.DataFrame(
+        {
+            "floor_to_day": [datetime(2008, 2, 15), datetime(2017, 8, 18)],
+        }
+    )
+    df1 = c.sql("SELECT FLOOR(to_timestamp(d) TO DAY) AS floor_to_day FROM df")
+    assert_eq(df1, expected_df)
+    df2 = c.sql("SELECT FLOOR(CAST(d AS TIMESTAMP) TO DAY) AS floor_to_day FROM df")
+    assert_eq(df2, expected_df)
+
+    df = pd.DataFrame({"d1": [1203073300], "d2": [1503073700]})
+    c.create_table("df", df, gpu=gpu)
+    expected_df = pd.DataFrame({"dt": [3472]})
+    df1 = c.sql(
+        "SELECT TIMESTAMPDIFF(DAY, to_timestamp(d1), to_timestamp(d2)) AS dt FROM df"
+    )
+    # TODO: The GPU case returns an incorrect value here
```

Comment:
Could you file an issue to track these failures on GPU (and the seconds/nanoseconds conversion issue below)?

Reply:
Yes, just opened https://github.com/dask-contrib/dask-sql/issues/1187. Thanks!